core/templates/components/smooth-height/smooth-height.component.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the smooth height change animation.
+ */
+
+import { ElementRef, HostBinding, Component, Input, OnChanges } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
+
+@Component({
+  selector: 'oppia-smooth-height',
```

Comment:
@ashutoshc8101 Any specific reason for creating this as a component and not directive (and use it as an attribute for a DOM/directive)?

To me `oppia-smooth-height` doesn't sound like a view instead it sounds like a attribute which can affect a view.

Reply:
because content projection can't be done using a directive.

Reply:
But do we really need the internal html in this directive/componenet? Can't we find the height using the directive? (not sure!)

Reply:
Angular animations don't work with directives. They only work with components.